File: Dynamic Programming/0122.Best Time to Buy and Sell Stock II/Solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        f0, f1 = 0, -prices[0]
        for i in range(1, n):
            tmp = f0
            f0 = max(f0, f1 + prices[i])
            f1 = max(f1, tmp - prices[i])
        return max(f0, f1)

File: Dynamic Programming/0122.Best Time to Buy and Sell Stock II/Solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int f0 = 0, f1 = -prices[0];
        for (int i = 1; i < prices.length; i++) {
            int tmp = f0;
            f0 = Math.max(f0, f1 + prices[i]);
            f1 = Math.max(f1, tmp - prices[i]);
        }
        return Math.max(f0, f1);
    }
}